source/type/type.cpp

```diff
@@ -554,6 +569,12 @@ const VehicleJourney* VehicleJourney::get_corresponding_base() const {
     return nullptr;
 }
 
+std::vector<StopTime>::const_iterator VehicleJourney::earliest_stop_time() const {
```

Comment:
Returning an iterator here is not so userfriendly: the end iterator to check is not trivial. Maybe returning a pointer or even a reference if that can't fail is a better idea.